CSharp/Library/Microsoft.Bot.Connector.Shared/ConnectorClientEx.cs

```diff
@@ -49,25 +49,44 @@ public ConnectorClient(Uri baseUri, MicrosoftAppCredentials credentials, HttpCli
         {
             this.Credentials = credentials;
         }
-        
+
         private static DelegatingHandler[] AddJwtTokenRefresher(DelegatingHandler[] srcHandlers, MicrosoftAppCredentials credentials)
         {
             var handlers = new List<DelegatingHandler>(srcHandlers);
             handlers.Add(new JwtTokenRefresher(credentials));
             return handlers.ToArray();
         }
-        
-        // client defaults to sending the expect: continue header, which isn't very efficient, 
+
+        private HttpClient instanceClient;
+        protected static HttpClient g_httpClient = null;
+        protected static object syncObj = new object();
+
         partial void CustomInitialize()
         {
-            AddUserAgent(this);
-            HttpClient.DefaultRequestHeaders.ExpectContinue = false;
+            if (g_httpClient == null)
+            {
+                lock (syncObj)
+                {
+                    if (g_httpClient == null)
+                    {
+                        g_httpClient = new HttpClient();
+                        g_httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("Microsoft-BotFramework", "3.1"));
+                        g_httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue($"(BotBuilder .Net/{GetClientVersion(this)})"));

```

Comment:
If this builds one long User-Agent string ("Microsoft-BotFramework/3.1 (BotBuilder .Net/1.2.3.4)" then awesome. Does it?